Documentation for Users endpoint related objects (#6515)
ppy/osu-web: resources/docs/source/includes/_structures.md

-----------|-----------------|--------------------------------------------
beatmapset | EventBeatmapset | |
user       | EventUser       | Beatmapset owner.

#### beatmapsetUpdate

When a beatmapset is updated.

Field      | Type            | Description
-----------|-----------------|--------------------------------------------
beatmapset | EventBeatmapset | |
user       | EventUser       | Beatmapset owner.

#### beatmapsetUpload

When a new beatmapset is uploaded.

Field      | Type            | Description
-----------|-----------------|--------------------------------------------
beatmapset | EventBeatmapset | |
user       | EventUser       | Beatmapset owner.

#### rank

When a user obtained certain rank on a beatmap.

Field     | Type         | Description
----------|--------------|--------------------------------------------
scoreRank | string       | (FIXME)
rank      | number       | |
mode      | GameMode     | |
beatmap   | EventBeatmap | |
user      | EventUser    | |

#### rankLost

When a user lost first place to another user.

Field     | Type
----------|-------------
mode      | GameMode
beatmap   | EventBeatmap
user      | EventUser

#### userSupportAgain

> When a user becomes supporter for the second or more time.

Reviewer: ```suggestion
When a user supports osu! for the second time and onwards.
```

Field     | Type
----------|----------
user      | EventUser

#### userSupportFirst

> When a user becomes supporter for the first time.

Reviewer: ```suggestion
When a user becomes a supporter for the first time.
```

Field     | Type
----------|----------
user      | EventUser
#### userSupportGift

> When a user received supporter tag gift from another user.

Reviewer: ```suggestion
When a user if gifted a supporter tag from another user.
```
